### ISSUE TITLE
After Migration from Magento 1.9.4 to 2.3.1 Customer can't login

### ISSUE THREAD

**Issue by cse00241** [magento/magento2#26669]
**Summary**
I recently Migrate to Magento 1.9.4 to Magento 2.3.1. But Some of my customers can't log in to there account. I did found the issue why it happen:
`php -f bin/magento customer:hash:upgrade`
This command used but it's not working.
user: mahbub0047@gmail.com does not work cause it has the password has like $2y$10$PTpXmxtV1U4MrGOCs302..naVdKUgwBlXVNykc504nVfrHn9.TLr6:0

But a user from old version magento did work fine cause it has like: 914f122d6b829b371ca9122b74528a0d3f7d6d5b924bb6bf309d171c850cc446:0o1ICsF2WcIcqIVeT8S2E72ggi9jzSVy:1

**Preconditions**
Magento 2.3.2

**Expected result**
I want to log in that customer who have the hash with $2y$10$. But not edit the core encryptor.php file.
https://i.stack.imgur.com/FdoXN.png

**Actual result**
My Magento Version is 1.9.4 and updated to 2.3.1.
After migration Magento 1.9.4 to Magneto 2.3.1 Using Uber Theme Can't Login Customer Account.
php -f bin/magento customer:hash:upgrade
This command used but its not working.

user : mahbub0047@gmail.com not work cause it has password has like: $2y$10$PTpXmxtV1U4MrGOCs302..naVdKUgwBlXVNykc504nVfrHn9.TLr6:0

But a user from old version magento did work fine cause it has like: 914f122d6b829b371ca9122b74528a0d3f7d6d5b924bb6bf309d171c850cc446:0o1ICsF2WcIcqIVeT8S2E72ggi9jzSVy:1

https://i.stack.imgur.com/FdoXN.png

I did fix it by edit Encryptor.php
```

public function isValidHash($password, $hash)

    {
        $password_old = $password;
        try {
            $this->explodePasswordHash($hash);
            foreach ($this->getPasswordVersion() as $hashVersion) {
                if ($hashVersion === self::HASH_VERSION_ARGON2ID13) {
                    $recreated = $this->getArgonHash($password, $this->getPasswordSalt());
                } else {
                    $recreated = $this->generateSimpleHash($this->getPasswordSalt() . $password, $hashVersion);
                }
                $hash = $this->getPasswordHash();
            }
        } catch (\RuntimeException $exception) {
            //Hash is not a password hash.
            $recreated = $this->hash($password);
        }

        $first_password = substr($hash, 0, 7);
        $hash_first = explode(":", $hash)[0];
        if (password_verify($password_old, $hash_first) && $first_password = '$2y$10$') {
            $recreated = $hash;
        }
        
        return Security::compareStrings(
            $recreated,
            $hash
        );
    }
```
But want to fix it with out edit core files.

**Comment by Fidelity88** [comment]
I'm having the same issue. The weird thing is, the password hash seems to be changed for recent customers.

As in: old customers (can't pinpoint exactly what "old" means, could be months, could be weeks) have a password hash that **ends with** **":1"** or **":0"**. But more recent customers, so customer that either recently purchased something OR recently updated/reset their password have a hash that **begins with** **"$2y$"**

For me the issue occurred with the migration of Magento 1.9.4.1 to 2.3.3

**Comment by engcom-Alfa** [comment]
Duplicate of #26731

**Comment by victor-v-rad** [comment]
@cse00241 You can try to use this fix magento/data-migration-tool@3c7f83b clean cache and empty magento_root/generated directory. When customers login the tool will validate the password and rehash it in algorithm which Magento 2 understand.
